pkg/cmd/certregenerationcontroller/cabundlesyncer.go

```diff
@@ -0,0 +1,148 @@
+	queue workqueue.RateLimitingInterface
+}
+
+func NewCABundleController(
+	configMapGetter corev1client.ConfigMapsGetter,
+	kubeInformersForNamespaces v1helpers.KubeInformersForNamespaces,
+	eventRecorder events.Recorder,
+) (*CABundleController, error) {
+	c := &CABundleController{
+		configMapGetter: configMapGetter,
+		configMapLister: kubeInformersForNamespaces.InformersFor(metav1.NamespaceAll).Core().V1().ConfigMaps().Lister(),
+		eventRecorder:   eventRecorder.WithComponentSuffix("manage-client-ca-bundle-recovery-controller"),
+		queue:           workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "CABundleRecoveryController"),
+	}
+
+	c.cachesToSync = append(c.cachesToSync, kubeInformersForNamespaces.InformersFor(metav1.NamespaceAll).Core().V1().ConfigMaps().Informer().HasSynced)
+
+	handler := cache.ResourceEventHandlerFuncs{
+		AddFunc:    func(obj interface{}) { c.queue.Add(workQueueKey) },
+		UpdateFunc: func(old, new interface{}) { c.queue.Add(workQueueKey) },
+		DeleteFunc: func(obj interface{}) { c.queue.Add(workQueueKey) },
+	}
+
+	// we react to some config changes
+	namespaces := []string{
+		operatorclient.GlobalUserSpecifiedConfigNamespace,
+		operatorclient.GlobalMachineSpecifiedConfigNamespace,
+		operatorclient.OperatorNamespace,
+		operatorclient.TargetNamespace,
+	}
+	for _, namespace := range namespaces {
+		informers := kubeInformersForNamespaces.InformersFor(namespace)
+		informers.Core().V1().ConfigMaps().Informer().AddEventHandler(handler)
+		c.cachesToSync = append(c.cachesToSync, informers.Core().V1().ConfigMaps().Informer().HasSynced)
+	}
+
+	return c, nil
+}
+
+func (c *CABundleController) Run(ctx context.Context) {
+	defer utilruntime.HandleCrash()
+
+	var wg sync.WaitGroup
+	klog.Info("Starting CA bundle controller")
+	defer func() {
+		klog.Info("Shutting down CA bundle controller")
+		c.queue.ShutDown()
+		wg.Wait()
+		klog.Info("CA bundle controller shut down")
+	}()
+
+	if !cache.WaitForNamedCacheSync("CABundleController", ctx.Done(), c.cachesToSync...) {
+		return
+	}
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		wait.UntilWithContext(ctx, c.runWorker, time.Second)
+	}()
+
+	<-ctx.Done()
+}
+
+func (c *CABundleController) runWorker(ctx context.Context) {
+	for c.processNextItem(ctx) {
+	}
+}
+
+func (c *CABundleController) processNextItem(ctx context.Context) bool {
+	key, quit := c.queue.Get()
+	if quit {
+		return false
+	}
+	defer c.queue.Done(key)
+
+	err := c.sync(ctx)
+
+	if err == nil {
+		c.queue.Forget(key)
+		return true
+	}
+
+	utilruntime.HandleError(fmt.Errorf("%v failed with : %w", key, err))
+	c.queue.AddRateLimited(key)
+
+	return true
+}
+
+func (c *CABundleController) sync(ctx context.Context) error {
+	// Always start 10 seconds later after a change occurred. Makes us less likely to steal work and logs from the operator.
+	select {
+	case <-time.NewTimer(10 * time.Second).C:
```

Comment:
time.After please.  You forgot to stop this one.

Reply:
`time.After` seems equivalent, I think I should rather stop it